mne/chpi.py

```diff
@@ -264,24 +270,52 @@ def _fit_chpi_pos(coil_dev_rrs, coil_head_rrs, x0):
     return x, 1. - result / denom
 
 
+def _fit_dev_head_trans(dev_pnts, head_pnts):
+    """Compute Device to Head transform allowing for permutiatons of points."""
+    import itertools
+    id_quat = np.concatenate([rot_to_quat(np.eye(3)), [0.0, 0.0, 0.0]])
+    best_order = None
+    best_g = -999
+    best_quat = id_quat
+    for i in itertools.permutations(np.arange(len(head_pnts))):
+        head_pnts_tmp = head_pnts[np.array(i)]
+        this_quat, g = _fit_chpi_quat(dev_pnts, head_pnts_tmp, id_quat)
+        if g > best_g:
+            best_g = g
+            best_order = np.array(i)
+            best_quat = this_quat
+
+    # Convert Quaterion to transform
+    dev_head_t = np.concatenate(
+        (quat_to_rot(best_quat[:3]),
+         best_quat[3:][:, np.newaxis]), axis=1)
+    dev_head_t = np.concatenate((dev_head_t, [[0, 0, 0, 1.]]))
+    return dev_head_t, best_order
```

Comment:
Can you add one to the master and I'll add to/maintain it through this process. That way the format will be what you want it to be from the start.